### PR TITLE
Modeling users

### PR DESCRIPTION
**Pull request by cupatea**

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -1,9 +1,7 @@
 class HomeController < ApplicationController
   def index
     @user = User.new
-  end
-
-  def show
+    @users = Fake.limit(12)
```
Comment (verkligheten) [inline app/controllers/home_controller.rb]: @users -> @fake_users

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -13,10 +11,16 @@ def privacy_policy
   end
 
   def search
+    _redirect_unless_current_user
+    @users = Fake.page params[:page]
```
Comment (verkligheten) [inline app/controllers/home_controller.rb]: @users -> @fake_users

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+class UsersController < ApplicationController
+   before_action :authenticate_user!
+
+   def show
+     @user = Fake.find_by_id params[:id]
```
Comment (verkligheten) [inline app/controllers/users_controller.rb]: @users -> @fake_users

File: app/views/home/profile.html.erb [hunk 4]
```diff
@@ -1,363 +1,382 @@
...
-                  <%= label_tag :email, t('email'), class:  "col-sm-2 col-form-label" %>
-                  <div class="col-sm-10">
-                    <%= text_field_tag :email, current_user.email, class: "form-control-plaintext", readonly: true %>
+
+            <div class="col-12 col-lg-8">
+
+              <div class="row mt-lg-5">
+                <div class="col">
+                  <div class="form-group row">
+                    <%= label_tag :email, t('email'), class:  "col-sm-2 col-form-label" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label

File: app/views/home/profile.html.erb [hunk 5]
```diff
@@ -1,363 +1,382 @@
...
-                    <div class="form-check form-check-inline">
-                      <%= radio_button_tag :gender, 'male', true, class: "form-check-input" %>
-                      <%= label_tag :male, t('male'), class:  "form-check-label" %>
+                  <div class="form-group row">
+                    <%= f.label :gender, t('gender'), class: "col-sm-2 col-form-label" %>
+                    <div class="col-sm-10 d-flex align-items-center">
+                      <% User.genders.keys.each do |gender| %>
+                        <div class="form-check form-check-inline">
+                          <%= f.radio_button :gender, gender, class: "form-check-input" %>
+                          <%= label_tag gender, t(gender), class:  "pl-2 form-check-label" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label

File: app/views/home/profile.html.erb [hunk 6]
```diff
@@ -1,363 +1,382 @@
...
-                </div>
-                <div class="form-group row">
-                  <%= label_tag :country, t('country'), class: "col-sm-2 col-form-label" %>
-                  <div class="col-sm-10">
-                    <%= select_tag :country,
-                      options_for_select([["Oslo", 0], ["Roagaland", 1], ["Oppland", 2]]),
-                      class: "custom-select"
-                    %>
+                  <div class="form-group row">
+                    <%= label_tag :language, t('language'), class: "col-sm-2 col-form-label" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label

File: app/views/home/profile.html.erb [hunk 7]
```diff
@@ -1,363 +1,382 @@
...
+                      <%= f.text_field :job, class: "form-control" %>
+                    </div>
                   </div>
-                </div>
-                <div class="form-group row">
-                  <%= label_tag :about, t('about'), class: "col-sm-2 col-form-label"%>
-                  <div class="col-sm-10">
-                    <%= text_area_tag :about, "", class: "form-control" %>
+                  <div class="form-group row">
+                    <%= label_tag :education, t('education'), class: "col-sm-2 col-form-label" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label

File: app/views/home/profile.html.erb [hunk 8]
```diff
@@ -1,363 +1,382 @@
...
-                  <div class="form-group row">
-                    <%= label_tag "activity#{index}", t("activity#{index}"), class: "col-sm-2 col-form-label" %>
-                    <div class="col-sm-10">
-                      <%= select_tag :country,
-                        options_for_select([["I'll tell you later", 0], ["Or", 1], ["Not", 2]]),
-                        class: "custom-select"
-                      %>
+              <div class="row">
+                <div class="col-12 col-lg-6">
+                  <% [ :hight_range, :eye_color, :hair_color, :haircut].each do |value|  %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: replace this array to User::SOME_GLOBAL_VAR

File: app/views/home/profile.html.erb [hunk 9]
```diff
@@ -1,363 +1,382 @@
...
-                    <%= label_tag "activity#{index}", t("activity#{index}"), class: "col-sm-2 col-form-label" %>
-                    <div class="col-sm-10">
-                      <%= select_tag :country,
-                        options_for_select([["I'll tell you later", 0], ["Or", 1], ["Not", 2]]),
-                        class: "custom-select"
-                      %>
+                  <% end %>
+                </div>
+                <div class="col-12 col-lg-6">
+                  <% [ :skin_color, :body_shape, :intimate_haircut, current_user.gender == 'male' ? :penis_size : :cup_size ].each do |value|  %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: replace this with instance method

File: app/views/home/profile.html.erb [hunk 10]
```diff
@@ -1,363 +1,382 @@
...
-                  %>
-                </div>
-                <div class="form-check form-check-inline">
-                  <%= label_tag :to, t('to'), class:  "form-check-label mr-2" %>
-                  <%= select_tag :education,
-                    options_for_select([18,20,25,30,35,40,45,50,55,60,65,70,75,80]),
-                    class: "form-control"
-                  %>
+              <div class="form-group row">
+                <%= label_tag :looking, t('looking'), class:  "col-sm-2 col-form-label" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label 
(remove spaces)

File: app/views/home/profile.html.erb [hunk 11]
```diff
@@ -1,363 +1,382 @@
...
+                <div class="col-sm-10 d-flex align-items-center">
+                  <div class="form-check form-check-inline">
+                    <%= radio_button_tag :gender, 'man', false, class: "form-check-input" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label 
f.radio

File: app/views/home/profile.html.erb [hunk 12]
```diff
@@ -1,363 +1,382 @@
...
-                <%= label_tag :country, t('country'), class:  "col-sm-2 col-form-label" %>
-                <div class="form-group col-sm-10 d-flex align-items-center mb-0">
-                  <div class="form-check-inline">
-                    <%= check_box_tag "country0", "country0", [true,false].sample, class: "form-check-input" %>
-                    <%= label_tag "country0", "#{('A'..'Z').to_a.sample}uzanding", class:  "form-check-label" %>
+              <div class="form-group row">
+                <%= label_tag :age, t('age'), class:  "col-sm-2 col-form-label" %>
+                <div class="col-sm-10">
+                  <div class="form-check form-check-inline">
+                    <%= label_tag :from, t('from'), class:  "form-check-label mr-2" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label

File: app/views/home/profile.html.erb [hunk 13]
```diff
@@ -1,363 +1,382 @@
...
               </div>
-              <% 4.times do |index| %>
-                <div class="form-group row">
-                  <div class="form-group col-sm-10 offset-sm-2 d-flex align-items-center mb-0">
-                    <div class="form-check form-check-inline">
-                      <%= check_box_tag "country#{index + 1}", "country#{index + 1}", [true,false].sample, class: "form-check-input" %>
-                      <%= label_tag "country#{index + 1}", "#{('A'..'Z').to_a.sample}uzanding", class:  "form-check-label" %>
+              <div id="country">
+                <div class="form-group mb-2 row">
+                  <%= label_tag :country, t('country'), class:  "col-sm-2 col-form-label" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.label

File: app/views/home/profile.html.erb [hunk 14]
```diff
@@ -1,363 +1,382 @@
...
+                  <div class="form-group col-sm-10 d-flex align-items-center mb-0">
+                    <div class="form-check-inline">
+                      <%= check_box_tag "country0", "country0", [true,false].sample, class: "form-check-input" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: country0??

File: app/views/home/profile.html.erb [hunk 15]
```diff
@@ -1,363 +1,382 @@
...
+                </div>
+              </div>
+            </div>
           </div>
-        </div>
+        </div> <!-- Search preferences end -->
         <div class="row">
           <div class="col d-flex justify-content-center my-4">
-            <%= button_tag t('save'), class: "wb-general__btn_red" %>
+            <%= f.button t('save'), class: "wb-general__btn_red" %>
```
Comment (verkligheten) [inline app/views/home/profile.html.erb]: f.submit

File: app/views/home/search.html.erb [hunk 16]
```diff
@@ -1,41 +1,56 @@
...
         <%= image_tag "separators/w-title-b.png", class: "mb-4" %>
       </div>
     </div>
     <div class="row">
-      <% 3.times do |index| %>
-        <div class="col-sm col-xs-6 mt-1">
-          <h4 class="text-white mb-1">Param<%= index + 1 %></h4>
-          <%= select_tag "option#{index}",
-            options_for_select([["Any", 0], ["One", 1], ["Two", 2]]),
+      <% [ :province, :intimate_haircut, current_user.gender == 'male' ? :penis_size : :cup_size ].each do |prop| %>
```
Comment (verkligheten) [inline app/views/home/search.html.erb]: replace this logic to model

File: app/views/kaminari/_paginator.html.erb [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+      <%= prev_page_tag unless current_page.first? %>
+      <% each_page do |page| %>
+        <% if page.left_outer? || page.right_outer? || page.inside_window? %>
+          <%= page_tag page %>
+        <% elsif !page.was_truncated? -%>
+          <%= gap_tag %>
+        <% end %>
+      <% end %>
+      <%= next_page_tag unless current_page.last? %>
+      <!-- <%= last_page_tag unless current_page.last? %> -->
```
Comment (verkligheten) [inline app/views/kaminari/_paginator.html.erb]: remove commented code

File: app/views/users/show.html.erb [hunk 18]
```diff
@@ -57,47 +57,55 @@
...
           </ul>
           <div class="tab-content">
-            <div role="tabpanel" class="tab-pane active show fade" id="activity">
+            <div role="tabpanel" class="tab-pane active show fade" id="about">
               <div class="wb-users__profile-information_list">
-                <% 2.times do %>
                   <ul>
-                    <% 4.times do  %>
-                      <li><%= "Activity title" %></li>
+                    <% [:relationship, :language, :job, :education, :smoking].each  do |prop|  %>
```
Comment (verkligheten) [inline app/views/users/show.html.erb]: replace logic

File: app/views/users/show.html.erb [hunk 19]
```diff
@@ -57,47 +57,55 @@
...
+                      <li><%= prop.to_s.humanize %></li>
                     <% end %>
                   </ul>
                   <ul>
-                    <% 4.times do  %>
-                      <li><%= "Activity info" %></li>
+                    <% [:relationship, :language, :job, :education, :smoking].each  do |prop| %>
```
Comment (verkligheten) [inline app/views/users/show.html.erb]: replace logic

File: app/views/users/show.html.erb [hunk 20]
```diff
@@ -57,47 +57,55 @@
...
               </div>
             </div>
-            <div role="tabpanel" class="tab-pane fade" id="profile">
+            <div role="tabpanel" class="tab-pane fade" id="appearance">
               <div class="wb-users__profile-information_list">
-                <% 2.times do %>
                   <ul>
-                    <% 4.times do  %>
-                      <li><%= "Profile title" %></li>
+                    <% [ :hight_range, :eye_color, :hair_color, :haircut].each do |prop|  %>
```
Comment (verkligheten) [inline app/views/users/show.html.erb]: replace logic

File: app/views/users/show.html.erb [hunk 21]
```diff
@@ -57,47 +57,55 @@
...
+                      <li><%= prop.to_s.humanize %></li>
                     <% end %>
                   </ul>
                   <ul>
-                    <% 4.times do  %>
-                      <li><%= "Profile info" %></li>
+                    <% [ :hight_range, :eye_color, :hair_color, :haircut].each do |prop|  %>
```
Comment (verkligheten) [inline app/views/users/show.html.erb]: replace logic

File: app/views/users/show.html.erb [hunk 22]
```diff
@@ -57,47 +57,55 @@
...
+                      <li><%= @user.public_send(prop) %></li>
+                    <% end %>
+                  </ul>
+                  <ul>
+                    <% [ :skin_color, :body_shape, :intimate_haircut, @user.gender == 'male' ? :penis_size : :cup_size ].each do |prop|  %>
```
Comment (verkligheten) [inline app/views/users/show.html.erb]: replace logic

File: app/views/users/show.html.erb [hunk 23]
```diff
@@ -57,47 +57,55 @@
...
+                      <li><%= prop.to_s.humanize %></li>
+                    <% end %>
+                  </ul>
+                  <ul>
+                    <% [ :skin_color, :body_shape, :intimate_haircut, @user.gender == 'male' ? :penis_size : :cup_size ].each do |prop|  %>
```
Comment (verkligheten) [inline app/views/users/show.html.erb]: replace logic

File: config/initializers/kaminari_config.rb [hunk 24]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+Kaminari.configure do |config|
+  config.default_per_page = 24
```
Comment (verkligheten) [inline config/initializers/kaminari_config.rb]: per_page = 20

File: app/views/home/profile.html.erb (hunk 15 above)
Comment (cupatea) [reply app/views/home/profile.html.erb]: can't do this, provided styles (wb-general__btn_red) are not applied to 'input' (f.submit), but only to 'button'(f.button)

**Pull request merged**